### DISCUSSION TITLE
Missing images in an article of Android Authority

### DISCUSSION THREAD

**Discussion by PhilC813** [jocmp/capyreader#502] [Q&A] [answered]
I am subscribed to the feed of Android Authority in Capy. I have the "sticky full content" option turned on so as to always extract the full content on feeds such as this one.

Here's the feed URL:
https://www.androidauthority.com/feed/

The vast majority of articles have images loading correctly. However, in the following article, which has side-by-side comparisons of images taken from a smartphone, the images aren't fetched by Capy:
https://www.androidauthority.com/zerocam-ai-3498885/

Would these be expected to be displayed in Capy?

I checked the page source code and I would assume that it's due to the more complex implementation in the HTML. It has the following tag structure for each image:
button > picture > source > img

**Comment by jocmp** [comment, marked as answer]
Thanks for digging into this. I believe you're right about the `button` tag. In the library Capy uses it looks like there's a "clean buttons" call ([source code](https://github.com/dankito/Readability4J/blob/170d052e99db58ecac85a77cdaa63ac8253be1fd/src/main/kotlin/net/dankito/readability4j/processor/ArticleGrabber.kt#L791)) which would explain why they're omitted. The library is a native fork of Firefox's reader which means it has some similar behavior. I loaded up that article in Firefox to compare and it also omitted the side-by-side images.
 
![Screenshot 2024-11-13 at 5 09 09 PM](https://github.com/user-attachments/assets/7f6913cf-6711-459d-9243-fd6d07e26cdf)

**Reply by PhilC813** [reply]
I'll check with their project if they have an open issue about this. Maybe they could add a condition to prevent some tag removals if they contain an image. Thank you as always Josiah!

**Reply by jocmp** [reply]
Sure thing! This button problem is similar to another ticket someone submitted to me https://github.com/jocmp/capyreader/issues/501. That Readability4J library itself is stable but inactive. I may fork it or find a different library at some point to improve image handling.

**Reply by PhilC813** [reply]
> That Readability4J library itself is stable but inactive.

Yeah that's what I realized... But the author does have GitHub activity so we never know!

I created an issue in their repo just in case:
https://github.com/dankito/Readability4J/issues/26

Forking or finding an alternative could indeed be a good idea.

**Comment by PhilC813** [comment]
That's another feed (in French sorry) for which extracting the full content can't fetch the featured image of the articles:
https://www.versants.com/feed/

It's like this for all the articles, but if you take this one for example:
https://www.versants.com/actualite/saint-bruno-le-sapin-de-noel-geant-est-installe-a-la-place-du-village/

That's the HTML of the image:
```
<figure class="mb-2 aspect-3/2 h-auto w-full overflow-hidden">
  <img width="1000" height="667" src="data:image/svg+xml,%3Csvg%20xmlns='http://www.w3.org/2000/svg'%20viewBox='0%200%201000%20667'%3E%3C/svg%3E" class="object-cover w-full h-full wp-post-image" alt="" decoding="async" data-lazy-srcset="https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-1000x667.jpg 1000w, https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-300x200.jpg 300w, https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-1536x1024.jpg 1536w, https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-768x512.jpg 768w" data-lazy-sizes="(max-width: 1000px) 100vw, 1000px" data-lazy-src="https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-1000x667.jpg" />
  <noscript>
    <img width="1000" height="667" src="https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-1000x667.jpg" class="object-cover w-full h-full wp-post-image" alt="" decoding="async" srcset="https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-1000x667.jpg 1000w, https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-300x200.jpg 300w, https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-1536x1024.jpg 1536w, https://www.versants.com/wp-content/uploads/2024/11/f986303d06476bd86303d064741963v_-768x512.jpg 768w" sizes="(max-width: 1000px) 100vw, 1000px" />
  </noscript>
</figure>
```

Is there some issue (i.e., #506) that could be used to gather all examples? If you ever fork the Readability4J library or change to another, it would probably really help if you had a central list with all affected feeds to benchmark the new solution.

**Reply by jocmp** [reply]
I like that idea. I pinned #506 and I'l track new feeds there. 

Instead of forking Readability4J, I'm considering forking a library called [Postlight Parser](https://github.com/postlight/parser). I initially used this in Capy because you can fix individual websites without changing the underlying parser. If I can solve that, using it would be a good fit for both the Android Authority example and this `<noscript>` example in Les Versants too.

**Reply by PhilC813** [reply]
Sweet! That will definitely help in edge cases. I hope it's not too difficult to implement.
